test/TransportTests.cpp

```diff
@@ -0,0 +1,243 @@
+
+  // Step 2: Create host-side array of device pointers
+  ncclDevChannelPeer* devPeerPtrsHost[3] = {
+    devPeerArrayDevice + 0,
+    devPeerArrayDevice + 1,
+    devPeerArrayDevice + 2
+  };
+
+  // Step 3: Allocate device-side array of device pointers
+  ncclDevChannelPeer** devPeerPtrsDevice;
+  hipMalloc(&devPeerPtrsDevice, sizeof(ncclDevChannelPeer*) * 3);
+
+  // Step 4: Copy host-side array of device pointers to device
+  hipMemcpy(devPeerPtrsDevice, devPeerPtrsHost, sizeof(ncclDevChannelPeer*) * 3, hipMemcpyHostToDevice);
+
+  // Step 5: Set in channel
+  channel.devPeers = devPeerPtrsDevice;
+
+  // --- Setup transportComm ---
+  static struct ncclTransportComm dummyTransport = {
+    .setup = mockSetup,
+    .connect = mockConnect,
+  };
+  collNetTransport.send = dummyTransport;
+
+  // --- Dummy inputs ---
+  ncclTopoGraph topoGraph = {};
+  ncclConnect connect = {};  // IMPORTANT: non-null since this is memcpy’d into masterConnects
+  int masterRank = 0;
+  int masterPeer = 1;
+  int channelId = 0;
+  int type = collNetSend;
+
+  // --- Run the function ---
+  bool failed = ncclTransportCollNetSetup(&comm, &topoGraph, &channel, masterRank, masterPeer, channelId, type, &connect);
+
+  // --- Assert: function should succeed (return false) ---
+  ASSERT_FALSE(failed);
+
+  // --- Cleanup ---
+  hipFree(devPeerArrayDevice);
+  hipFree(devPeerPtrsDevice);
+}
+
+
+  TEST(TransportTest, NcclTransportCollNetCheckTestSuccess) {
+    struct ncclComm comm = {};
+    struct ncclBootstrap dummyBootstrap;
+
+    int rankMap[1] = {0};
+    comm.localRank = 0;
+    comm.localRanks = 1;
+    comm.localRankToRank = rankMap;
+    comm.bootstrap = &dummyBootstrap;
+    int collNetSetupFail = 0;
+    ncclResult_t result = ncclTransportCollNetCheck(&comm, collNetSetupFail);
+    EXPECT_EQ(result, ncclSuccess);
+  }
+
+  TEST(TransportTest, NcclTransportCollNetCheckTestFails) {
+    ncclComm comm;
+    int rankMap[1] = {0};
+    ncclBootstrap bootstrap;
+
+    comm.localRank = 0;
+    comm.localRanks = 1;
+    comm.localRankToRank = rankMap;
+    comm.bootstrap = &bootstrap;
+
+    int collNetSetupFail = 1; // simulate failure on this rank
+    ncclResult_t result = ncclTransportCollNetCheck(&comm, collNetSetupFail);
+    EXPECT_EQ(result, ncclSystemError);
+}
+
+// Test for ncclTransportCollNetFree
+TEST(TransportTest, CollNetFreeTest) {
+  ncclComm comm = {};
+  comm.nChannels = 1;
+  comm.nRanks = 0; // So comm.channels[0].peers[0] is accessed
+
+  // Access embedded array directly (don't assign to comm.channels)
+  ncclChannel* channel = &comm.channels[0];
+
+  // Allocate peer array for the channel (comm.nRanks + 1 for dummy peer)
+  channel->peers = new ncclChannelPeer*[comm.nRanks + 1];
+  for (int r = 0; r <= comm.nRanks; ++r) {
+    channel->peers[r] = nullptr;
+  }
+
+  // Create dummy peer at index nRanks
+  ncclChannelPeer* peer = new ncclChannelPeer();
+  channel->peers[comm.nRanks] = peer;
```

Comment:
You could construct on the stack instead of managing heap allocation.
```suggestion
  ncclChannelPeer peer;
  channel->peers[comm.nRanks] = &peer;
```

Reply:
Thanks for the suggestions — they’re all fair points. I’ve reviewed it again and I believe the PR is good to go. It meets the criteria and the tests are meaningful. Any remaining gaps can be handled in follow-ups.